src/store/recordAction.ts

```diff
@@ -0,0 +1,13 @@
+import * as FullStory from '@fullstory/browser';
+import { Middleware } from 'redux';
+
+const recordAction: Middleware = () => next => action => {
+  if (typeof action !== 'function') {
+    FullStory.log(action.type);
+  } else if (typeof action === 'function') {
+    FullStory.log('Action is a thunk');
+  }
+  return next(action);
```

Comment:
I know you've put a lot of thought into how crash reporting should be handled. One way that this can be tested with redux actions by [catching the exception here, and rethrowing again](https://redux.js.org/advanced/middleware#problem-crash-reporting). 

How do you see exception handling fitting into this? Outside of this scope?

Reply:
Great question! I see it outside of the scope at the moment.

In an earlier reference app (the one I used for Sentry) I have two middlewares, one for handling errors and the other for recording actions: https://github.com/patrick-fs/fs-react-redux-sentry/blob/master/src/store/index.js

Reply:
That makes sense to me!